streaming/src/test/scala/org/apache/spark/streaming/ReceivedBlockHandlerSuite.scala

```diff
@@ -62,6 +61,19 @@ class ReceivedBlockHandlerSuite
   var blockManagerMaster: BlockManagerMaster = null
   var blockManager: BlockManager = null
   var tempDirectory: File = null
+  var storageLevel = StorageLevel.MEMORY_ONLY_SER
```

Comment:
Aah I see what you have done. It was val but you made it a var so that it can be modified to set a different storage level which `withXYZBasedBlockHandler` can use. This is non-intuitive and changes global state which can affect other tests. Not a good idea. Rather make the storageLevel an *optional* parameter in `withXYZBasedBlockHandler` and the default value being the earlier global value `StorageLevel.MEMORY_ONLY_SER`. Then we dont even need to have the global variable `storageLeval` at all.